Experiments/forest_QKS/qks.py

```diff
@@ -0,0 +1,223 @@
+import numpy as np
+import matplotlib.pyplot as plt
+
+from sklearn.linear_model import LogisticRegression
+from sklearn.utils import shuffle
+
+from pyquil import get_qc
+from pyquil.quil import Program
+from pyquil.gates import RX, CNOT, MEASURE
+
+
+def make_frames(num):
+    """
+    The picture frames dataset contains a smaller inner square
+    with sides of length 2 and a larger outer square with sides
+    of length 4. Both have points uniformally distributed around
+    the average.
+    """
+    def zip_arrays(d1,d2):
+        return np.array(list(zip(d1,d2)))
+
+    def make_squares(num, l):
+        left_right, top_bottom = make_barriers(num,l), make_barriers(num,l)
+        top_bottom[:,[0,1]] = top_bottom[:,[1,0]] 
+        square = np.concatenate((left_right, top_bottom))
+        return square
+
+    def make_barriers(num, l):
+        dist = 0.1*l
+        left = np.random.uniform((-l-dist),(-l+dist),num)
+        right = np.random.uniform((l-dist),(l+dist),num)
+        left_barrier = np.random.uniform(-l,l,num)
+        right_barrier = np.random.uniform(-l,l,num)
+        L = zip_arrays(left,left_barrier)
+        R = zip_arrays(right,right_barrier)
+        barriers = np.vstack((L,R))
+        return barriers
+    
+    outer = make_squares(num, l=2)
+    inner = make_squares(num, l=1)
+    frames = np.concatenate((outer,inner))
+    
+    half = int((len(frames)/2))
+    target = np.concatenate((np.zeros(half),np.ones(half)))
+    
+    X,y = shuffle(frames, target, random_state=1337)
+    return X,y
+
+
+def make_omega(n_episodes,p,q,r,scale,loc=0):
+    """
+    encodes the input vector into q gate parameters
+    """
+
+    def create_selection_matrix(p,q,r):
+        """
+        Generates a matrix of 0s and 1s to zero out `r` values per matrix
+        """
+        matrix_size = p*q
+        m = np.zeros(matrix_size)
+        for i in range(r): m[i] = 1
+        np.random.shuffle(m)
+        selection_matrix = m.reshape(q,p)
+        return selection_matrix
+
+    norm_dist = np.random.normal(loc, scale, size=(n_episodes,q,p))
```

Comment:
norm means "normal" here, and not "normalized" right?

Reply:
Yes, I'll rename for clarification.